c19-backend/ehrbase_connector/connector.py

```diff
@@ -13,5 +14,71 @@ def connect(*, base_url) -> 'OpenEHRConnector':
 class OpenEHRConnector(object):
     base_url = attr.ib()
 
-    def get(self, path):
-        return requests.get(self.base_url + path)
+    def _url(self, suffix):
+        return self.base_url + suffix
+
+    def get(self, *, path, params=None, **kwargs):
+        # TODO also pass auth=(user, pass) once basic auth implemented
+        return requests.get(self._url(path), params=None, **kwargs)
+
+    def post(self, *, path, data=None, json=None, **kwargs):
+        # TODO also pass auth=(user, pass) once basic auth implemented
+        return requests.post(self._url(path), data=data, json=json, **kwargs)
+
+
+@attr.s(frozen=True)
+class OpenEHRAPI(object):
+    connector = attr.ib()
+
+    def ehr_id_from_nhs_number(self, *, nhs_number: str):
+        def ehr_already_existed(status_code):
+            return status_code == 409  # Conflict
+        nhs_number_namespace = 'uk.nhs.number'
+        creation_response = self.connector.post(
+            '/ehrbase/rest/openehr/v1/ehr',
+            json={
+                "_type": "EHR_STATUS",
+                "subject": {
+                    "external_ref": {
+                        "id": {
+                            "_type": "GENERIC_ID",
+                            "value": nhs_number,
+                            "scheme": "uk.nhs.nhs_number"
+                        },
+                        "namespace": nhs_number_namespace,
+                        "type": "PERSON"
+                    }
+                },
+                "is_modifiable": "true",
+                "is_queryable": "true",
+            },
+        )
+        if (
```

Comment:
if condition:
   body
no 
if (condition):
  body

You can break the line by '\' if condition is to long:
if a == True and \
   b == False